topo/node/lemming/lemming.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package lemming contains a node implementation for a lemming device.
+package lemming
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/openconfig/kne/topo/node"
+
+	tpb "github.com/openconfig/kne/proto/topo"
+)
+
+func New(nodeImpl *node.Impl) (node.Node, error) {
+	if nodeImpl == nil {
+		return nil, fmt.Errorf("nodeImpl cannot be nil")
+	}
+	if nodeImpl.Proto == nil {
+		return nil, fmt.Errorf("nodeImpl.Proto cannot be nil")
+	}
+	cfg := defaults(nodeImpl.Proto)
+	nodeImpl.Proto = cfg
+	n := &Node{
+		Impl: nodeImpl,
+	}
+	return n, nil
+}
+
+type Node struct {
+	*node.Impl
+}
+
+// Add validations for interfaces the node provides
+var (
+	_ node.Certer       = (*Node)(nil)
+	_ node.ConfigPusher = (*Node)(nil)
+	_ node.Resetter     = (*Node)(nil)
+)
+
+func (n *Node) ResetCfg(ctx context.Context) error {
+	return nil
+}
+
+func (n *Node) ConfigPush(context.Context, io.Reader) error {
+	return fmt.Errorf("config push is not implemented using gNMI to configure device")
```

Comment:
can you make these return status.Unimplemented grpc error codes?

Reply:
done